### PR TITLE
[CAZ-1723] Update Payment Status - when EntrantPayment is missing

### PR DESCRIPTION
**Pull request by musztardem**
**Introduced changes:**
* When request `PUT /payment-status` tries to update `EntrantPayment` which does not exist - the record is being created with the proper data.

**Additional info:**
* [Link to CAZ-1723 JIRA Card](https://eaflood.atlassian.net/browse/CAZ-1723)

### CODE REVIEW

File: src/main/java/uk/gov/caz/psr/service/PaymentStatusUpdateService.java [hunk 1]
```diff
@@ -81,6 +82,27 @@ private EntrantPayment prepareUpdateEntrantPayment(EntrantPayment entrantPayment
...
+   * @param statusUpdate {@link EntrantPaymentStatusUpdate} object used to initialize new
+   *     record.
+   * @return entrantPayment {@link EntrantPayment} which is supposed to be persisted in the
+   *     database.
+   */
+  private EntrantPayment prepareCreateEntrantPayment(EntrantPaymentStatusUpdate statusUpdate) {
+    return EntrantPayment.builder()
+        .vrn(statusUpdate.getVrn())
+        .cleanAirZoneId(statusUpdate.getCleanAirZoneId())
+        .travelDate(LocalDate.now())
```
Comment (radekmorytko) [inline src/main/java/uk/gov/caz/psr/service/PaymentStatusUpdateService.java]: I think that `travelDate` should be equal to `uk.gov.caz.psr.model.EntrantPaymentStatusUpdate#dateOfCazEntry`

File: src/main/java/uk/gov/caz/psr/service/PaymentStatusUpdateService.java [hunk 2]
```diff
@@ -1,6 +1,7 @@
 package uk.gov.caz.psr.service;
 
 import com.google.common.base.Preconditions;
+import java.time.LocalDate;
```
Comment (radekmorytko) [inline src/main/java/uk/gov/caz/psr/service/PaymentStatusUpdateService.java]: Could you please add some logging with `INFO` level to this class? I would expect some message for every case with some parameters.

File: src/main/java/uk/gov/caz/psr/service/PaymentStatusUpdateService.java [hunk 3]
```diff
@@ -41,13 +41,15 @@ public void process(
...
       } else {
-        handleNewEntrantPayment();
+        handleNewEntrantPayment(entrantPaymentStatusUpdate);
       }
     }
   }
 
-  private void handleNewEntrantPayment() {
-    // TODO: Implement in CAZ-1723
+  private void handleNewEntrantPayment(EntrantPaymentStatusUpdate entrantPaymentStatusUpdate) {
```
Comment (plenza) [inline src/main/java/uk/gov/caz/psr/service/PaymentStatusUpdateService.java]: `public process()` method should be `@Transactional`

File: src/main/java/uk/gov/caz/psr/service/PaymentStatusUpdateService.java [hunk 4]
```diff
@@ -81,6 +86,27 @@ private EntrantPayment prepareUpdateEntrantPayment(EntrantPayment entrantPayment
...
 
+  /**
+   * Builds {@link EntrantPayment} with assigned default attributes.
+   *
+   * @param statusUpdate {@link EntrantPaymentStatusUpdate} object used to initialize new
+   *     record.
+   * @return entrantPayment {@link EntrantPayment} which is supposed to be persisted in the
+   *     database.
+   */
+  private EntrantPayment prepareCreateEntrantPayment(EntrantPaymentStatusUpdate statusUpdate) {
```
Comment (plenza) [inline src/main/java/uk/gov/caz/psr/service/PaymentStatusUpdateService.java]: `prepareCreateEntrantPayment` - Maybe rename this to `makeNewEntrantPayment(EntrantPaymentStatusUpdate .. )`?

File: src/main/java/uk/gov/caz/psr/service/PaymentStatusUpdateService.java [hunk 5]
```diff
@@ -81,6 +86,27 @@ private EntrantPayment prepareUpdateEntrantPayment(EntrantPayment entrantPayment
         .build();
   }
 
+  /**
+   * Builds {@link EntrantPayment} with assigned default attributes.
```
Comment (plenza) [inline src/main/java/uk/gov/caz/psr/service/PaymentStatusUpdateService.java]: I would state it:
"Builds new instance of {@link EntrantPayment} with attributes assigned by Local Authority, with zero charge"

File: src/test/java/uk/gov/caz/psr/service/PaymentStatusUpdateServiceTest.java [hunk 6]
```diff
@@ -112,19 +113,52 @@ public void shouldUpdateWithProvidedData() {
...
+
+    // when
+    paymentStatusUpdateService.process(entrantPaymentStatusUpdates);
+
+    // then
+    verify(entrantPaymentRepository).insert(expectedEntrantPayment);
+  }
 
-  private void mockVehicleEntrantPaymentFoundWith(EntrantPayment entrantPayment) {
+  private EntrantPayment buildExpectedEntrantPaymentFrom(EntrantPaymentStatusUpdate statusUpdate) {
```
Comment (plenza) [inline src/test/java/uk/gov/caz/psr/service/PaymentStatusUpdateServiceTest.java]: If you had a component in prod section (not tests) that did this conversion, you could eliminate code duplication between [this](https://github.com/InformedSolutions/JAQU-CAZ-Payments-API/pull/128/files#diff-1dd367fcfdf0bf1aa8d66269aad8387bR98) and test here and by bonus have it tested by this test.

File: src/it/java/uk/gov/caz/psr/SuccessPaymentStatusUpdateTestIT.java [hunk 7]
```diff
@@ -96,6 +104,13 @@ private PaymentStatusUpdateRequest paymentStatusUpdateRequest(String vrn) {
         .build();
   }
 
+  private PaymentStatusUpdateRequest paymentStatusUpdateRequestForNonExistingParams() {
+    return PaymentStatusUpdateRequest.builder()
+        .vrn("MARYSIA")
```
Comment (radekmorytko) [inline src/it/java/uk/gov/caz/psr/SuccessPaymentStatusUpdateTestIT.java]: Nice

**Pull request merged**